core/domain/state_domain_test.py

```diff
@@ -637,6 +639,587 @@ def test_validate_state_solicit_answer_details(self):
+                },
+                'labelled_as_correct': False,
+                'param_changes': [],
+                'refresher_exploration_id': None,
+                'missing_prerequisite_skill_id': None
+            },
+            'rule_specs': [{
+                'inputs': {
+                    'x': 'Test'
+                },
+                'rule_type': 'Contains'
+            }],
+            'training_data': [],
+            'tagged_misconception_id': None
+        }
+
+        exploration.init_state.update_interaction_answer_groups(
+            [answer_groups_dict])
+        exploration.init_state.update_content({
+            'content_id': 'feedback_1',
+            'html': 'Feedback'
+        })
+
+        with self.assertRaisesRegexp(Exception, 'Found a duplicate content id'):
+            exploration.init_state.validate(None, True)
+
+    def test_validate_duplicate_content_id_with_default_outcome(self):
+        exploration = self.save_new_valid_exploration('exp_id', 'owner_id')
+        default_outcome_dict = {
+            'dest': 'Introduction',
+            'feedback': {
+                'content_id': 'default_outcome',
+                'html': ''},
+            'labelled_as_correct': False,
+            'missing_prerequisite_skill_id': None,
+            'param_changes': [],
+            'refresher_exploration_id': None
+        }
+
+        exploration.init_state.update_interaction_default_outcome(
+            default_outcome_dict)
+        exploration.init_state.update_content({
+            'content_id': 'default_outcome',
+            'html': ''
+        })
+
+        with self.assertRaisesRegexp(Exception, 'Found a duplicate content id'):
+            exploration.init_state.validate(None, True)
+
+    def test_validate_duplicate_content_id_with_hints(self):
+        exploration = self.save_new_valid_exploration('exp_id', 'owner_id')
+        hints_list = [{
+            'hint_content': {
+                'content_id': 'hint_1',
+                'html': 'some html'
+            }
+        }]
+
+        exploration.init_state.update_interaction_hints(hints_list)
+        exploration.init_state.update_content({
+            'content_id': 'hint_1',
+            'html': ''
+        })
+
+        with self.assertRaisesRegexp(Exception, 'Found a duplicate content id'):
+            exploration.init_state.validate(None, True)
+
+    def test_validate_duplicate_content_id_with_solution(self):
+        exploration = self.save_new_valid_exploration('exp_id', 'owner_id')
+        subtitled_html = state_domain.SubtitledHtml('content_id', 'some html')
+
+        hints_list = [state_domain.Hint(subtitled_html)]
+
+        exploration.init_state.interaction.hints = hints_list
+        solution = {
+            'answer_is_exclusive': True,
+            'correct_answer': 'hello_world!',
+            'explanation': {
+                'content_id': 'solution',
+                'html': 'hello_world is a string'
```

Comment:
I think we are expecting to have all the text to be inside an html tag (like `<p>...</p>`).

/cc @ankita240796 is that correct?

Reply:
Yes, that's correct.

Reply:
done